feat: yield new application programming interface

raisedadead opened the pull request:
Umbrella API PR. Do not merge or push commits here directly.

ShaunSHamilton committed:
refactor(api): shiny new api (#48432)

ShaunSHamilton committed:
feat(api): example middleware (#48434)

* feat(api): add middleware example

* feat(api): add `@fastify/middie`, reorder for alphabetness

* [skip-ci] [skip ci]

ShaunSHamilton committed:
feat(api): add node:test (#48437)

ojeytonwilliams committed:
refactor: clean up types and tsconfig (#49360)

raisedadead commented:
Let's get this in to avoid all the conflicts with the lock file. The new API will live in its folder, so we should be OK working on the `main`.

raisedadead commented:
The tests are failing because `jest` is picking up the `.test.ts` files in the API folder; it needs to be ignored for now (if we do not intend on using `jest`).

@ShaunSHamilton how do you want to proceed?

ShaunSHamilton commented:
> The tests are failing because `jest` is picking up the `.test.ts` files in the API folder; it needs to be ignored for now (if we do not intend on using `jest`).
> 
> @ShaunSHamilton how do you want to proceed?

#49371

ShaunSHamilton committed:
fix(api): ignore api in jest config (#49371)

ojeytonwilliams committed:
refactor: clean up errors + unused code (#49473)

* refactor: use await/done to mollify eslint

* refactor: clean up unused code

It's also more of an express pattern. Fastify routes should be
registered as plugins.

raisedadead committed:
feat(next-api): basic authentication setup (#49378)

* feat(next-api): add fastify-auth0-verify plugin

* feat(next-api): add fastify-jwt-authz plugin

* feat(next-api): accept privacy endpoint with scopes support

* fix(next-api): ignore eslint and ts errors

They will be fixed in a future PR when the package with errors has been updated

Co-authored-by: Niraj Nandish <nirajnandish@icloud.com>
Co-authored-by: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>
